consensus/src/main/java/org/apache/iotdb/consensus/ratis/SnapshotStorage.java

```diff
@@ -166,11 +167,14 @@ public void format() throws IOException {}
   @Override
   public void cleanupOldSnapshots(SnapshotRetentionPolicy snapshotRetentionPolicy)
       throws IOException {
-    Path[] sortedSnapshotDirs = getSortedSnapshotDirPaths();
-    if (sortedSnapshotDirs == null || sortedSnapshotDirs.length == 0) {
+    final Path[] sortedSnapshotDirs = getSortedSnapshotDirPaths();
+    if (ArrayUtils.isEmpty(sortedSnapshotDirs)) {
       return;
     }
-    for (int i = 0; i < sortedSnapshotDirs.length - 1; i++) {
+
+    final int retainIndex =
+        snapshotRetentionPolicy.getNumSnapshotsRetained() - 1; // snapshot rank starts from 0
+    for (int i = retainIndex; i < sortedSnapshotDirs.length - 1; i++) {
```

Comment:
The modification code and master branch are different. why?

Reply:
Fixed. Now they are the same. Thanks!